homeassistant/components/deconz/lock.py

```diff
@@ -72,16 +86,18 @@ class DeconzLock(DeconzDevice, LockEntity):
     """Representation of a deCONZ lock."""
 
     TYPE = DOMAIN
+    _device: DoorLock | Lock
 
     @property
-    def is_locked(self):
+    def is_locked(self) -> bool:
         """Return true if lock is on."""
+        assert isinstance(self._device.is_locked, bool)
```

Comment:
Given that _device is type and is_locked is typed to return a bool, why is this assert needed? It seems like there should be enough information to infer this type without that, right?

Reply:
This is due to the library not being used to trace types, I haven't added py.typed to it yet since that would require a lot of more rewrites, so Im taking it in steps.

Reply:
Fix the library instead. `ininstance` checks are runtime. This now adds runtime load for typing. Which IMHO is not a good solution.

Reply:
Based on feedback in other PRs I'm working through everything and removing all asserts

Reply:
Fixed